flow/core/kernel/power_demand.py

```diff
@@ -16,49 +16,38 @@ class PowerDemandModel(BaseEnergyModel, metaclass=ABCMeta):
 
     def __init__(self,kernel):
         self.k = kernel
-        self.k.env.vehicle = vehicle
         self.g = 9.81
         self.rho_air = 1.225
         self.mass = 1200
         self.rolling_res_coeff = 0.005
-        self.aerodymnamic_drag_coeff = 0.3
+        self.aerodynamic_drag_coeff = 0.3
         self.cross_area = 2.6
         self.gamma = 1
     
-    def calculate_power(self):
-        speed = self.k.env.get_speed(veh_id)
-        if veh_id in self.k.env.previous_speeds:
-            old_speed = self.k.env.previous_speeds[veh_id]
-        else:
-            old_speed = speed
-
-        accel = (speed - old_speed)/self.k.env.sim_step
-        accel_slope_forces = self.mass * speed * (
-                     (np.heaviside(accel, 0.5) * (1 - self.gamma) + self.gamma)) * accel
-                     + self.g * math.sin(grade))
+    def calculate_power(self, accel, speed, grade):
+        accel_slope_forces = self.mass * speed * ((np.heaviside(accel, 0.5) * (1 - self.gamma) + self.gamma)) * accel + self.g * math.sin(grade)

```

Comment:
i think the parens should be like this:
`self.gamma) * accel + self.g * math.sin(grade))`

also this line is too long, so reformat it into 2-3 lines